src/handlers/LiquidityHandler.sol

```diff
@@ -364,13 +395,37 @@ contract LiquidityHandler is OwnableUpgradeable, ReentrancyGuardUpgradeable, ILi
     if (msg.sender != address(this)) revert ILiquidityHandler_Unauthorized();
 
     if (_order.isAdd) {
-      IERC20Upgradeable(_order.token).safeTransfer(LiquidityService(liquidityService).vaultStorage(), _order.amount);
-      _amountOut = LiquidityService(liquidityService).addLiquidity(
-        _order.account,
-        _order.token,
-        _order.amount,
-        _order.minOut
-      );
+      if (address(hlpStaking) != address(0)) {
+        // If HLPStaking is live
+        IERC20Upgradeable(_order.token).safeTransfer(LiquidityService(liquidityService).vaultStorage(), _order.amount);
+        _amountOut = LiquidityService(liquidityService).addLiquidity(
+          _order.account,
+          _order.token,
+          _order.amount,
+          _order.minOut,
+          address(this)
+        );
+
+        // Auto stake into HLPStaking
+        IERC20Upgradeable(ConfigStorage(LiquidityService(liquidityService).configStorage()).hlp()).safeApprove(
+          address(hlpStaking),
+          _amountOut
+        );
+        if (isHyper[_order.account][_order.orderId]) {
+          IHyperStaking(hlpStaking).depositHyper(_order.account, _amountOut);
+        } else {
+          IHyperStaking(hlpStaking).deposit(_order.account, _amountOut);
+        }
+      } else {
+        // If no HLPStaking
+        IERC20Upgradeable(_order.token).safeTransfer(LiquidityService(liquidityService).vaultStorage(), _order.amount);
+        _amountOut = LiquidityService(liquidityService).addLiquidity(
```

Comment:
you can combine addLiquidity part to 1 block and have ternary to decide whether which address to receive the HLP

Reply:
I'm gonna keep it as is. Changing this, make old test failed.